### ISSUE TITLE
Question: Why the following code is invalid?

### ISSUE THREAD

**Issue by jsynowiec** [zertosh/eslint-plugin-no-async-without-await#1]
I'm trying to use this rule in my project but even the simplest function is reported as invalid.

**Function declaration**

``` javascript
function delayedHello(name: string, delay: number = 2000): Promise<string> {
  return new Promise((resolve) => setTimeout(resolve(`Hello, ${name}`), delay));
}

async function greeter(name: string): Promise<string> {
  const hello = await delayedHello(name);
  return hello;
}
```

**ESLint output**

```
7:1  error  Unnecessary async (only use async when await is needed)  no-async-without-await/no-async-without-await
```

**.eslintrc**

``` json
{
  "extends": "airbnb-base",
  "parser": "babel-eslint",
  "plugins": [
    "flowtype",
    "no-async-without-await"
  ],
  "rules": {
    "no-async-without-await/no-async-without-await": 2
  }
}

```

Libraries
- babel-eslint 7.0.0
- eslint 3.8.0

@zertosh can you please explain why and how to fix this?

**Comment by zertosh** [comment]
eslint 3.7.0 introduced builtin support for `async`/`await` so the AST is a little different. I'll publish a fix later today. Sorry.

**Comment by jsynowiec** [comment]
Support for `async/await` was introduced in eslint 3.6.0 ;-) Ok, I'll move back to some earlier version for now and wait for the fix. Thanks!

Maybe it's something else, because even on 3.5.0 I have same results.

**Comment by zertosh** [comment]
Just published `eslint-plugin-no-async-without-await@1.2.0` that supports both the new and old babel-eslint

**Comment by zertosh** [comment]
Fixed by 2f96046cc96c29b9486eb4c3fd96c4e24223b83a
